solutions/usaco-941.mdx

```diff
@@ -143,4 +174,68 @@ public class CowEvolution {
 ```
 
 </JavaSection>
+
+<CPPSection>
+
+```cpp
+// Taken from official analysis by Brian Dean
+#include <iostream>
+#include <fstream>
+#include <vector>
+using namespace std;
+ 
+int N;
+vector<string> characteristics[25];
+vector<string> all_characteristics;
+ 
+// Do two sets "cross" -- I.e., are there elements in A, B, and A intersect B?
+bool crossing(int a, int b)
+{
+	int A = 0, B = 0, AB = 0;
+	for (int i = 0; i < N; i++) {
+	vector<string> &v = characteristics[i];
+	bool has_a = false, has_b = false;
+	for (int j = 0; j < v.size(); j++) {
+	  if (v[j] == all_characteristics[a]) has_a = true;
+	  if (v[j] == all_characteristics[b]) has_b = true;
+	}
+	if (has_a && has_b) AB++;
+	else if (has_a) A++;
+	else if (has_b) B++;
+	}
```

Comment:
use tabs instead of spaces to indent
and i think the indentation is messed up too